<launch>
    <!-- Orbbec Astra Pro behind the robot -->
    <node name="usb_cam_behind" pkg="usb_cam" type="usb_cam_node" output="screen" >
        <rosparam command="load" file="$(find larcc_bringup)/config/camera_behind.yml" />
    </node>

    <!-- Object Recognition Pipeline - Image to Label -->
    <include file="$(find pamaral_object_grasping_pattern_recognition)/launch/object_recognition_pipeline.launch">
        <arg name="input_image_topic" value="usb_cam_behind/image_raw"/>
    </include>

    <!-- Visualization -->
    <node name="data_visualization" pkg="pamaral_object_grasping_pattern_recognition" type="data_visualization.py" output="screen">
        <param name="input_image_topic" value="usb_cam_behind/image_raw"/>
    </node>

    <!-- Test -->
    <node name="test_node" pkg="pamaral_object_grasping_pattern_recognition" type="object_recognition_pipeline_test.py" output="screen"/>

    <!-- RViz -->
    <node name="rviz" pkg="rviz" type="rviz"
          args="--display-config $(find pamaral_object_grasping_pattern_recognition)/config/rviz.rviz" required="false"/>
</launch>